### DISCUSSION TITLE
Struggling with mod_oauth2 and Azure AD

### DISCUSSION THREAD

**Discussion by mshaddock** [OpenIDC/mod_oauth2#57] [Q&A]
I'm struggling with mod_oauth2 and Azure AD. I will admit that I don't know much about OAuth2, but should I set OAuth2TokenVerify to something like

OAuth2TokenVerify jwks_uri https://login.microsoftonline.com/common/discovery/keys

Any help is greatly appreciated.

**Comment by GRRedWings** [comment]
MS had a couple of different schemes for their endpoints, I use

OAuth2TokenVerify jwks_uri https://login.microsoftonline.com/${AZURE_TENANT_ID}/discovery/v2.0/keys

**Reply by mshaddock** [reply]
Thank you, that was exactly what I was looking for.
